src/plugins/createPluginManager.ts

```diff
@@ -0,0 +1,28 @@
+import {Plugin, PluginSlots} from '@plugins/PluginsContext';
+
+export class PluginManager {
+  private plugins: Plugin[];
+
+  public constructor() {
+    this.plugins = [];
+  }
+
+  public add(plugin: Plugin): void {
+    plugin.order ? this.plugins.splice(plugin.order, 0, plugin) : this.plugins.push(plugin);
+  }
+
+  public setup(): PluginSlots {
+    const pluginSlots: PluginSlots = {
+      executionDetailsTabs: [],
+    };
+    this.plugins.forEach(plugin => {
+      plugin.setup(pluginSlots);
```

Comment:
Instead of getting raw objects, `plugin.setup` should get some utility to help to set it up, or this object would need to have helper methods.

As an example:
```ts
class PluginsConfiguration {
  public slots: PluginSlots = {
    executionDetailsTabs: [],
  };

  addTo(slotName: keyof PluginSlots, order?: number): void;
  addTo(slotName: string, order?: number): void;
  addTo(slotName: string, order?: number): void {
    // add an item to slot
  },
}

// [...]

public setup(): PluginsConfiguration {
  const config = new PluginsConfiguration();
  this.plugins.forEach(plugin => {
    plugin.setup(config);
  });
  return config;
}
```

Reply:
refactored the data model exactly like that, to have a render function, that can receive props, which can be passed to the components. This will be a very common use case, as we will need to pass contextual data each time (e.g. selected test execution, etc).

Reply:
To pass contextual data, the components should use contexts instead; it's good to add support for providers too then